src/receiver.rs

```diff
@@ -19,22 +20,32 @@ pub enum Receiver {
 }
 
 impl Receiver {
-    pub fn parse(input: &str) -> Result<Self> {
+    pub fn parse(input: &str, auth_type: &AuthType) -> Result<Self> {
+        log::debug!("Parsing receiver: {}", input);
+        // Attempt to parse the input as a URL
         match Url::parse(input) {
             Ok(url) => {
-                let es_receiver = ElasticsearchReceiver::new(url);
-                return Ok(Self::Elasticsearch(es_receiver));
+                let auth = Auth::new(
+                    auth_type,
+                    None,
```

Comment:
should we be filling these arguments with the contents of the corresponding env vars as well?